src/Send/Context/SendContext.tsx

```diff
@@ -0,0 +1,76 @@
+import React, {useEffect, useState} from 'react'
+import {createContext, useContext} from 'react'
+
+import {getPrimaryAssetByNetworkId} from '../../legacy/config'
+import {YoroiWallet} from '../../yoroi-wallets'
+import {TokenId, YoroiAmounts} from '../../yoroi-wallets/types'
+
+const SendContext = createContext<undefined | SendContext>(undefined)
+export const SendProvider: React.FC<SendContextProvider> = ({children, balances, wallet}) => {
+  const primaryTokenId = getPrimaryAssetByNetworkId(wallet.networkId).identifier
+
+  const [selectedTokenIdentifier, setSelectedTokenIdentifier] = useState<TokenId>(primaryTokenId)
+  const [sendAll, setSendAll] = useState(false)
```

Comment:
React.useState

Reply:
actually, all this should go into useReducer

Reply:
it looks like we are moving towards what `formik` package does, perhaps instead of creating our own ctx, we could leverage it.

Reply:
jtol

Reply:
useReducer applied. Thanks.